app/main.py

```diff
@@ -1,34 +1,118 @@
 class Deck:
-    def __init__(self, row, column, is_alive=True):
-        pass
+    def __init__(self, row: int, column: int, is_alive: bool = True) -> None:
+        self.rows = row
+        self.columns = column
+        self.is_alive = is_alive
+        self.game_deck = [["~" for _ in range(column)] for _ in range(row)]
+
+    def __getitem__(self, index: tuple) -> list:
+        row, col = index
+        if 0 <= row < self.rows and 0 <= col < self.columns:
+            return self.game_deck[row][col]
+
+    def __setitem__(self, index: tuple, value: str) -> None:
+        row, col = index
+        if 0 <= row < self.rows and 0 <= col < self.columns:
+            self.game_deck[row][col] = value
+
+    def __str__(self) -> str:
+        return "\n".join(" ".join(row) for row in self.game_deck)
 
 
 class Ship:
-    def __init__(self, start, end, is_drowned=False):
-        # Create decks and save them to a list `self.decks`
-        pass
+    def __init__(
+            self,
+            start: tuple,
+            end: tuple,
+            is_drowned: bool = False
+    ) -> None:
+        self.start = start
+        self.end = end
+        self.is_drowned = is_drowned
+        self.hits = 0
+
+    def put_ship(self, deck: Deck) -> Deck:
+        if self.start[0] == self.end[0]:
+            for x_coord in range(self.start[1], self.end[1] + 1):
+                deck[self.start[0], x_coord] = "□"
+        else:
+            for y_coord in range(self.start[0], self.end[0] + 1):
+                deck[y_coord, self.start[1]] = "□"
+
+        return deck
+
+    def get_deck(self, row: int, column: int) -> bool:
```

Comment:
You must return a deck I think here, and you will can check `is_alive` attribute of Deck